### PR TITLE
chore(ui-react-native): Handle dial codes

### PR DESCRIPTION
**Pull request by ioanabrooks**
#### Description of changes

This change removes the dial code picker due to:
* lack of control over the styling of the third party Picker component that is currently in use, and its default style does align with our theming/styling
* one way door if we release with it

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-native/src/Authenticator/hooks/useFieldValues/useFieldValues.ts [hunk 1]
```diff
@@ -89,10 +89,18 @@ export default function useFieldValues<FieldType extends TypedField>({
...
+                For phone numbers pass the first 3 charactes from value as dialCode until we support a dialCode picker
+            */
+          return type === 'phone'
+            ? {
+                ...acc,
+                country_code: value?.substring(0, 3),
+                [name]: value?.substring(3, value.length),
+              }
+            : { ...acc, [name]: value };
+        }, {});
```
Comment (calebpollman) [inline packages/react-native/src/Authenticator/hooks/useFieldValues/useFieldValues.ts:103]: Would consider creating a meaningfully named utility for the callback, and explicitly using the `phone_number` key in place of `[name]` for the `phone` handling to be clear about what the keys are

File: packages/react-native/src/primitives/TextField/styles.ts [hunk 2]
```diff
@@ -19,23 +19,22 @@ export const getThemedStyles = (theme: StrictTheme): TextFieldStyles => {
...
-    inputContainer: {
+    fieldContainer: {
       alignItems: 'center',
       borderColor: colors.border.primary,
       borderRadius: radii.small,
       borderWidth: borderWidths.small,
       flexDirection: 'row',
       lineHeight: getLineHeight(fontSizes.medium),
       padding: space.medium,
-      width: '100%',
```
Comment (calebpollman) [inline packages/react-native/src/primitives/TextField/styles.ts:30]: Will this impact existing `TextField` usage?

**Review by calebpollman — COMMENTED** [review]
Approach works for me. Left some suggestions and a question

File: packages/react-native/src/primitives/TextField/styles.ts (hunk 2 above)
Comment (ioanabrooks) [reply packages/react-native/src/primitives/TextField/styles.ts:30]: I tested our authenticator screens and didn't see impact.

File: packages/react-native/src/Authenticator/hooks/useFieldValues/useFieldValues.ts (hunk 1 above)
Comment (ioanabrooks) [reply packages/react-native/src/Authenticator/hooks/useFieldValues/useFieldValues.ts:103]: Synced offline

**Review by calebpollman — APPROVED** [review]

**Review by zchenwei — APPROVED** [review]

**Pull request merged**